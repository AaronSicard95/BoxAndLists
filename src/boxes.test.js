import { fireEvent, queryByText, render } from "@testing-library/react";
import BoxForm from "./boxForm";
import BoxList from "./boxList";

it("check render", function(){
    render(<BoxForm/>);
});

it("match snap", function(){
    const {asFragment, getByLabelText, queryByText} = render(<BoxForm/>);
    expect(asFragment()).toMatchSnapshot();
    expect(queryByText("X")).not.toBeInTheDocument();
    fireEvent.click(queryByText("Add!"));
    expect(queryByText("X")).toBeInTheDocument();
})