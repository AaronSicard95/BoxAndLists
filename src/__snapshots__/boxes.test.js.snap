// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snap 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        name="width"
        type="number"
        value="50"
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        name="height"
        type="number"
        value="50"
      />
      <label
        for="color"
      >
        Color:
      </label>
      <input
        name="color"
        type="test"
        value="black"
      />
      <button>
        Add!
      </button>
    </form>
    <div
      style="display: inline;"
    />
  </div>
</DocumentFragment>
`;
